[Process] Fixes to executable finder tests

mpajunen opened the pull request:
| Q | A |
| --- | --- |
| Bug fix? | yes |
| New feature? | no |
| BC breaks? | no |
| Deprecations? | no |
| Tests pass? | yes |
| Fixed tickets | N/A |
| License | MIT |
| Doc PR | N/A |

A couple of fixes for ExecutableFinder tests:
- Use PHPUnit ini_set wrapper to avoid side effects.
- Skip a test on Windows (similar to an earlier test that was already skipped).
- Removed an unused variable.

mpajunen committed:
[Process] Removed unused variable assignment

mpajunen committed:
[Process] Added a test skip check for Windows

The test FindProcessInOpenBasedir fails similarly to
FindWithOpenBaseDir on Windows.

fabpot commented:
Replacing `ini_set` by `$this->initSet()` looks like a very good idea, can you do this change in all test files as I suppose it makes sense to always use it, right?

nicolas-grekas commented:
Double :+1:, one for the PR and one for using `$this->iniSet()` everywhere

mpajunen committed:
Use PHPUnit ini_set wrapper in tests

PHPUnit ini_set wrapper is now used in tests to automatically reset
ini settings after the test is run. This avoids possible side effects
and test skipping.

Native ini_set is still used in DefaultCsrfProviderTest, but its
tests are run in isolation.

mpajunen commented:
I replaced `ini_set()` with `$this->iniSet()` where applicable for all tests in 2.3. Would it be better to make a separate PR instead of using this one?

There are some tests in later versions that should probably be changed as well, not sure how that should be handled.

stof commented:
:+1:

fabpot commented:
Thank you @mpajunen.

fabpot commented:
@mpajunen Now that this PR is merged into all branches, can you submit a new PR for the iniSet() added in 2.5? Thanks.
